### PR TITLE
feat(mria): Add prometheus metrics

### PR DESCRIPTION
**Pull request by k32**

### CODE REVIEW

File: apps/emqx_prometheus/src/emqx_prometheus_mria.erl [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                                               )).
+
+%%====================================================================
+%% Internal functions
+%%====================================================================
+
+metrics() ->
+    Metrics = case mria_rlog:role() of
+                  replicant ->
+                      [lag, bootstrap_time, bootstrap_num_keys, message_queue_len];
```
Comment (thalesmg) [inline apps/emqx_prometheus/src/emqx_prometheus_mria.erl:69]:
```suggestion
                      [lag, bootstrap_time, bootstrap_num_keys, message_queue_len, replayq_len];
```
?

File: apps/emqx_prometheus/src/emqx_prometheus_mria.erl [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                  core ->
+                      [last_intercepted_trans, weight, replicants, server_mql]
+              end,
+    [{MetricId, fun() -> get_shard_metric(MetricId) end} || MetricId <- Metrics].
+
+get_shard_metric(Metric) ->
+    %% TODO: only report shards that are up
```
Comment (thalesmg) [inline apps/emqx_prometheus/src/emqx_prometheus_mria.erl:76]: Q: for this, couldn't we use `mria_status:shards_up/0`?

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_prometheus/src/emqx_prometheus_mria.erl (hunk 2 above)
Comment (k32) [reply apps/emqx_prometheus/src/emqx_prometheus_mria.erl:76]: Currently it returns an empty list on the core.

Comment (thalesmg) [reply apps/emqx_prometheus/src/emqx_prometheus_mria.erl:76]: I've just observed this right now, was about to open an issue to track it in mria. :hear_no_evil:

**Review by thalesmg — APPROVED** [review]

**Pull request merged**